lib/mox/server.ex

```diff
@@ -6,40 +6,46 @@ defmodule Mox.Server do
   # Public API
 
   def start_link(_options) do
-    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+    GenServer.start_link(__MODULE__, %{expectations: %{}, allowances: %{}}, name: __MODULE__)
   end
 
   def add_expectation(owner_pid, key, value) do
     GenServer.call(__MODULE__, {:add_expectation, owner_pid, key, value})
   end
 
-  def fetch_fun_to_dispatch(owner_pid, key) do
-    GenServer.call(__MODULE__, {:fetch_fun_to_dispatch, owner_pid, key})
+  def fetch_fun_to_dispatch(caller_pid, key) do
+    GenServer.call(__MODULE__, {:fetch_fun_to_dispatch, caller_pid, key})
   end
 
   def pending_expectations(owner_pid, for) do
     GenServer.call(__MODULE__, {:pending_expectations, owner_pid, for})
   end
 
+  def allow(mock, owner_pid, pid) do
+    GenServer.call(__MODULE__, {:allow, mock, owner_pid, pid})
+  end
+
   # Callbacks
 
   def handle_call({:add_expectation, owner_pid, key, expectation}, _from, state) do
```

Comment:
This function now needs to return {:error, :currently_allowed} if we are adding an expectation to an allowed process.

Reply:
Is this really a good idea? Maybe we should raise instead, or actually keep current behavior.

The reason why I think it might not be great idea is because I think it'll generally be used in pipe chain, and user won't detect error unless they have another call in the pipe operator, i.e.

```elixir
MyMock
|> allow(pid)
|> allow(:pid)
```

will return indeed {:error, :currently_allowed} but the user would only get an error if they had another line in the chain of |> and even then - the error might be confusing.

Reply:
ah @josevalim just saw the next comment, I was going one by one. Yes, I understand you want to raise error now - just from Mox api. Makes sense, sorry for the above.